### PR TITLE
Add error message for missing value for verbosity

### LINKED ISSUE #7520: dotnet restore -v throws ArgumentOutOfRangeException
## Steps to reproduce

Run the following in a Windows Command Prompt:

    dotnet restore -v

This can be done from anywhere and does not have to be, for example, run from a directory with a proper .NET Core project.

## Expected  behavior

The help for `dotnet restore` says:

```
Usage: dotnet restore [options]

Options:
  -h, --help                           Show help information.
  -s, --source <SOURCE>                Specifies a NuGet package source to use during the restore.
  -r, --runtime <RUNTIME_IDENTIFIER>   Target runtime to restore packages for.
  --packages <PACKAGES_DIRECTORY>      Directory to install packages in.
  --disable-parallel                   Disables restoring multiple projects in parallel.
  --configfile <FILE>                  The NuGet configuration file to use.
  --no-cache                           Do not cache packages and http requests.
  --ignore-failed-sources              Treat package source failures as warnings.
  --no-dependencies                    Set this flag to ignore project to project references and only restore the root project.
  -f, --force                          Set this flag to force all dependencies to be resolved even if the last restore was successful. This is equivalent to deleting project.assets.json.
  -v, --verbosity                      Set the verbosity level of the command. Allowed values are q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic].
```

The `-v` option is not indicated to have a value (e.g. it doesn't read `-v, --verbosity <LEVEL>`) so one would expect the normal level to be defaulted. The description, however, does say that it expects a value. If that's the case then one would expect an error along the lines of the option missing a value, similar to how running `dotnet restore -s` does by saying `Required argument missing for option: --source`.

## Actual behavior

The command fails with the following error (possibly due to an unhandled exception?):

```
System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
   at System.ThrowHelper.ThrowArgumentOutOfRange_IndexException()
   at System.SZArrayHelper.get_Item[T](Int32 index)
   at System.Linq.Enumerable.ElementAt[TSource](IEnumerable`1 source, Int32 index)
   at Microsoft.DotNet.Cli.Telemetry.TelemetryFilter.LogVerbosityForAllTopLevelCommand(ICollection`1 result, ParseResult parseResult, String topLevelCommandName)
   at Microsoft.DotNet.Cli.Telemetry.TelemetryFilter.Filter(Object objectToFilter)
   at Microsoft.DotNet.Cli.Utils.TelemetryEventEntry.SendFiltered(Object o)
   at Microsoft.DotNet.Cli.Program.ProcessArgs(String[] args, ITelemetry telemetryClient)
   at Microsoft.DotNet.Cli.Program.Main(String[] args)
```

## Environment data

`dotnet --info` output:

```
.NET Command Line Tools (2.0.0)

Product Information:
 Version:            2.0.0
 Commit SHA-1 hash:  cdcd1928c9

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.10586
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.0.0\

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.0
  Build    : e8b8861ac7faf042c87a5c2f9f2d04c98b69f28d
```

### PR DESCRIPTION
**Pull request by mikkelbu**
Fixes #7520

### CODE REVIEW

**Comment by mikkelbu** [comment]
@dotnet-bot
test CentOS7.1 x64 Debug Build please
test Debian8.2 x64 Debug Build please
test Linux x64 Release Build please
test Ubuntu16.04 x64 Debug Build please

**Comment by mikkelbu** [comment]
@dotnet-bot
test Debian8.2 x64 Debug Build please
test Linux x64 Release Build please

File: test/dotnet.Tests/TelemetryCommandTest.cs [hunk 1]
```diff
@@ -227,6 +229,16 @@ public void InternalreportinstallsuccessCommandCollectExeNameWithEventname()
         }
 
         [Fact]
+        public void TopLevelCommandFailsIfNoValueIsSuppliedAsVerbosity()
```
Comment (livarcocc) [inline test/dotnet.Tests/TelemetryCommandTest.cs]: I am not sure this test belongs in a TelemetryCommandTest class.

Comment (mikkelbu) [reply test/dotnet.Tests/TelemetryCommandTest.cs]: I thought that this was the best location available. Should I create a new file, called TelemetryFilterTests.cs, in the same folder as TelemetryCommandTest.cs, and then move the test to that file?

**Comment by wli3** [comment]
https://github.com/dotnet/cli/pull/7636

**Pull request closed without merging**